DPL Analysis: centralised CCDB support in analysis

ktf opened the pull request:
Thanks to the newly added binary view columns we can finally support
proper CCDB integration in analysis.

In order to do so, the user needs to create a TIMESTAMPED table, i.e. a table
which is an extension of another one where the timestamps for each rows
are provided. The extra columns of such timestamped table will be CCDB columns
where the iterator of each provides access for one specified CCDB object.

Notice the PR duplicates the CCDB code run in reconstruction and the additional
device will never be added to the topology if running online, so there are no expected
side effects.

alibuild commented:
Error while checking build/O2/fullCI_slc9 for a7b27956410df539633fc35800fe4b6b2c29c775 at 2025-08-10 18:36:
```
## sw/BUILD/O2-latest/log
[ERROR] invalid workflow in /sw/BUILD/05b14db3cfde306e932e5b730c7debb85c783acd/O2/stage/bin/o2-testworkflows-analysis-ccdb: In spec simple-ccdb-fetcher input specification 1 requires binding, description and origin to be fully specified (found :AOD/TOFCALIB/1)
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14567/a7b27956410df539633fc35800fe4b6b2c29c775/build_O2_fullCI_slc9/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI_slc9 for be4cb11972f838ce4695b7417ca0aa531db32afb at 2025-08-15 11:38:
```
## sw/BUILD/O2-latest/log
CMake Error at cmake/O2AddLibrary.cmake:76 (add_library):
CMake Error at cmake/O2AddLibrary.cmake:76 (add_library):
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14567/be4cb11972f838ce4695b7417ca0aa531db32afb/build_O2_fullCI_slc9/pretty.html).

ktf commented:
Still to do:

* actual fetching
* Making the fTimestamp column consigurable

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 35a19078471ad2784ea483d5cd74f9e3b84edf1c at 2025-08-16 23:54:
```
## sw/BUILD/O2-latest/log
CMake Error at cmake/O2AddLibrary.cmake:76 (add_library):
CMake Error at cmake/O2AddLibrary.cmake:76 (add_library):
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14567/35a19078471ad2784ea483d5cd74f9e3b84edf1c/build_O2_fullCI_slc9/pretty.html).

ktf commented:
With this the example (o2TestAnalysisCCDB.cxx) of the centralised CCDB for analysis tasks actually works. 

I need to do some cleanup before I can merge this, and I need to check I did not break the non analysis CCDB access, but we should not be far.

alibuild commented:
Error while checking build/O2/fullCI_slc9 for 71abb21ccb5144d23f25d351aada94ed12e0a6d9 at 2025-08-20 21:53:
```
++ cp /sw/slc9_x86-64/O2/14567-slc9_x86-64-local1/compile_commands.json .
+++ python3 -c 'import json, os; print(os.path.commonpath([x["file"] for x in json.loads(open("compile_commands.json").read()) if "sw/BUILD" not in x["file"] and "G__" not in x["file"] and x["file"].endswith(".cxx")]))'
++ O2_SRC=/sw/SOURCES/O2/14567-slc9_x86-64/0
++ [[ -e /sw/SOURCES/O2/14567-slc9_x86-64/0/CMakeLists.txt ]]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14567/71abb21ccb5144d23f25d351aada94ed12e0a6d9/build_O2_fullCI_slc9/pretty.html).

ktf committed:
DPL Analysis: centralised CCDB support in analysis

Thanks to the newly added binary view columns we can finally support
proper CCDB integration in analysis.

In order to do so, the user needs to create a TIMESTAMPED table, i.e. a table
which is an extension of another one where the timestamps for each rows
are provided. The extra columns of such timestamped table will be CCDB columns
where the iterator of each provides access for one specified CCDB object.

Notice the PR duplicates the CCDB code run in reconstruction and the additional
device will never be added to the topology if running online, so there are no expected
side effects.

ktf commented:
Despite the large size, this PR only introduces new features so it should be relatively safe to merge. In particular I took extra care in making sure the code is not enabled at all in online more and I duplicated anything which might have been used also by the reconstruction CCDB fetcher, to minimise risks. This has also been tested on HyperLoop with no regressions in Correlations and Histograms, as expected. Merging.
